front_end/content.py

```diff
@@ -2019,6 +2023,38 @@ def remove_export_paths(self, zip_file_path, tmp_dir_path):
         if os.path.exists(tmp_dir_path):
             shutil.rmtree(tmp_dir_path, ignore_errors=True)
 
+    def course_has_pair_programming(self, course_id):
+        sql = '''SELECT enable_pair_programming
```

Comment:
We can make this more efficient by only returning a single number. Please change `SELECT enable_pair_programming` to `SELECT MAX(enable_pair_programming)`. If any exercise has pair programming enabled, you will get a `1` back. Otherwise, you will get a 0 back.

Reply:
Then you can simplify the logic later in the function and use `fetchone`.